File: social-network-api/src/main/java/com/facebook/model/chat/ContentType.java
package com.facebook.model.chat;

public enum ContentType {
    TEXT,
    IMAGE,
    LIKE;
    public static ContentType fromString(String type) {
        return switch (type.toLowerCase()) {
            case "text" -> ContentType.TEXT;
            case "image" -> ContentType.IMAGE;
            case "like" -> ContentType.LIKE;
            default -> throw new IllegalArgumentException("Invalid ContentType: " + type);
        };
    }
}

File: social-network-ui/src/utils/contentType.js
export const ContentType = {
  TEXT: 'text',
  IMAGE: 'image',
  LIKE: 'like',
};

export function fromString(typeString) {
  switch (typeString.toLowerCase()) {
    case 'text':
      return ContentType.TEXT;
    case 'image':
      return ContentType.IMAGE;
    case 'like':
      return ContentType.LIKE;
    default:
      throw new Error('Invalid ContentType: ' + typeString);
  }
}

export function checkContentType(type, text) {
  if (fromString(type) === ContentType.IMAGE) {
    return "Надіслано світлину"
  } else if (fromString(type) === ContentType.LIKE) {
    return "Нравится!";
  } else {
    return text;
  }
}